recipes/mpc/all/conanfile.py

```diff
@@ -16,65 +23,78 @@ class MpcConan(ConanFile):
     default_options = {"shared": False, "fPIC": True}
     exports_sources = "patches/**"
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
 
     def config_options(self):
         if self.settings.os == 'Windows':
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
     def requirements(self):
-        self.requires("mpfr/4.1.0")
+        self.requires("gmp/6.2.1", transitive_headers=True)
+        self.requires("mpfr/4.1.0", transitive_headers=True)
 
     def validate(self):
-        # FIXME: add Visual Studio support, upstream has a makefile.vc
-        if self.settings.compiler == "Visual Studio":
-            raise ConanInvalidConfiguration("mpc can be built with Visual Studio, but it's not supported yet in this recipe.")
+        # FIXME: add msvc support, upstream has a makefile.vc
+        if is_msvc(self):
+            raise ConanInvalidConfiguration("mpc can be built with msvc, but it's not supported yet in this recipe.")
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        if self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
 
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        args = []
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append(f'--with-gmp={unix_path(self, self.dependencies["gmp"].package_folder)}')
+        tc.configure_args.append(f'--with-mpfr={unix_path(self, self.dependencies["mpfr"].package_folder)}')
         if self.options.shared:
-            args.extend(["--disable-static", "--enable-shared"])
+            tc.configure_args.extend(["--disable-static", "--enable-shared"])
```

Comment:
This "should" not be required. According to  https://docs.conan.io/en/2.0/reference/tools/gnu/autotoolstoolchain.html  it's managed automatically.


> Also if the shared option exists it will add by default:
> 
>         --enable-shared, --disable-static if shared==True
>         --disable-shared, --enable-static if shared==False

Reply:
Removed if/else clause as suggested.